IncompatibleClassChangeError opening in-memory graph under Java 9

## Steps to reproduce:

1. Create a project with dependency `org.janusgraph:janusgraph-core:0.2.0`
    - demo project: [janus-icce](https://github.com/dmolesUC3/janus-icce)
2. Under Java 9, run the following code:

```
    JanusGraph graph = JanusGraphFactory.build().set("storage.backend", "inmemory").open();
```

## Expected:

In-memory `JanusGraph` instance is a drop-in replacement for in-memory TinkerGraph (`TinkerGraph.open()`).

## Actual:

```
org.janusgraph.core.JanusGraphException: Could not execute operation due to backend exception

	at org.janusgraph.diskstorage.util.BackendOperation.execute(BackendOperation.java:57)

...

Caused by: org.janusgraph.diskstorage.PermanentBackendException: Permanent exception while executing backend operation getConfiguration
	at org.janusgraph.diskstorage.util.BackendOperation.executeDirect(BackendOperation.java:82)

...

Caused by: java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in classfile for class org/janusgraph/diskstorage/StaticBuffer. Method lambda$static$0([BII)[B at index 105 is CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
	at org.janusgraph.diskstorage.StaticBuffer.<clinit>(StaticBuffer.java:79)
	at org.janusgraph.diskstorage.configuration.backend.KCVSConfiguration$1.call(KCVSConfiguration.java:105)
	at org.janusgraph.diskstorage.configuration.backend.KCVSConfiguration$1.call(KCVSConfiguration.java:100)
	at org.janusgraph.diskstorage.util.BackendOperation.execute(BackendOperation.java:148)
```

(Full stack trace in a [gist here](https://gist.github.com/dmolesUC3/e6cdcbf0bcbcb1ba60678886e2c7f6d6).)

## Environment

OS: mac OS Sierra 10.12.6 (16G1114)

Java: 9.0.1+11

```
$ java -version
java version "9.0.1"
Java(TM) SE Runtime Environment (build 9.0.1+11)
Java HotSpot(TM) 64-Bit Server VM (build 9.0.1+11, mixed mode)
```

--- a/janusgraph-codepipelines-ci/pom.xml
+++ b/janusgraph-codepipelines-ci/pom.xml
@@ -83,7 +83,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>3.0.0</version>
+                <version>3.1.0</version>
                 <executions>
                     <execution>
                         <id>shade-codepipelines</id>

--- a/janusgraph-core/pom.xml
+++ b/janusgraph-core/pom.xml
@@ -196,7 +196,7 @@
             </plugin>
             <plugin>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>2.3</version>
+                <version>3.1.0</version>
                 <executions>
                     <execution>
                         <id>shade-gremlin-groovy</id>

